### DISCUSSION TITLE
Creating DB connection pools with lifespan

### DISCUSSION THREAD

**Discussion by csnatarajan** [fastapi/fastapi#9520] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
#utils.py
<releveant imports>
class DBPool:
    _connection_pool = None
    # conn_type will be 'read' or 'write'
    # In production connecting to Aurora Postgres. Local testing with Postgres container
    async def _create_connection_pool(conn_type: str):
        # Do a bunch of AWS specific stuff and
        DBPool._connection_pool = await asyncpg.create_pool(
            **credentials,
        )

# app.py
from .utils import DBPool
<other imports>
@asynccontextmanager
async def lifespan(app: FastAPI):
    await DBPool._create_connection_pool("read")
    yield
    # something here to tear down the pool
```

### Description

Is this a reasonable use of Lifespan to create a connection pool? I searched with Google and couldn't find a recommendation. Examples typically use the startup and shutdown events. It looks like the example in the FastAPI doc is similar to above. Looking for some guidance on the pattern/use of lifespan.

### Operating System

Linux

### Operating System Details

Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.2 LTS
Release:	22.04
Codename:	jammy

### FastAPI Version

0.94.0

### Python Version

3.10.9

### Additional Context

Trying to understand how to use lifespan to create a connection pool on startup and tear-down when done serving.

Deployment details:
1. App is bundled as pex with pants as build tool
2. Docker deployment on AWS

**Comment by chrisK824** [comment, marked as answer]
Hi there, haven't tested that out but it seems legit on a first look. As per example one needs only to pass lifespan function into fastAPI instance using the relat ed keyword lifespan. Also you can close your connection after the yield statement as this code is being executed after the app shutdowns.
